astral-sh/ruff-pre-commit
# Update mirror.py to a uv script

## Summary

From my reading of the [UV docs](https://docs.astral.sh/uv/guides/scripts/), `mirror.py` would be a good example of a standalone script. This PR updates `mirror.py` and the `main.yml` workflow to be a UV script and makes use of UV in the update process.

## Test Plan

- Update the uv version (currently 0.8.4) to an older version e.g. 0.8.3 in both `pyproject.toml` and `README.md`
- Run `uv run --no-project mirror.py` to check the update process still works as expected
